test_cluster_detection.py

```diff
@@ -0,0 +1,350 @@
+        assert (detections_in_world is not None)
+        assert (len(detections_in_world) == EXPECTED_CLUSTER_COUNT)
+
+    def test_detect_large_std_dev_single_cluster(self, cluster_model: ClusterEstimation):
+        """
+        data with large distribution and equal number of points per cluster center.
+        """
+        # Setup
+        POINTS_PER_CLUSTER = [100]
+        EXPECTED_CLUSTER_COUNT = len(POINTS_PER_CLUSTER)
+        generated_detections, _, _ = generate_cluster_data(POINTS_PER_CLUSTER,
+                                                           self.STD_DEV_LARGE)
+
+        # Run
+        model_ran, detections_in_world = cluster_model.run(generated_detections, False)
+
+        # Test
+        assert model_ran
+        assert (detections_in_world is not None)
+        assert (len(detections_in_world) == EXPECTED_CLUSTER_COUNT)
+
+    def test_detect_large_std_dev_five_clusters(self, cluster_model: ClusterEstimation):
+        """
+        data with large distribution and equal number of points per cluster center. 
+        """
+        # Setup
+        POINTS_PER_CLUSTER = [100, 100, 100, 100, 100]
+        EXPECTED_CLUSTER_COUNT = len(POINTS_PER_CLUSTER)
+        generated_detections, _, _ = generate_cluster_data(POINTS_PER_CLUSTER,
+                                                           self.STD_DEV_LARGE)
+
+        # Run
+        model_ran, detections_in_world = cluster_model.run(generated_detections, False)
+
+        # Test
+        assert model_ran
+        assert (detections_in_world is not None)
+        assert (len(detections_in_world) == EXPECTED_CLUSTER_COUNT)
+
+    def test_detect_skewed_data_single_cluster(self, cluster_model: ClusterEstimation):
+        """
+        data with small distribution but varying number of points per cluster center and
+        random outlier points to simulate false detections
+        """
+        # Setup
+        POINTS_PER_CLUSTER = [10, 100]
+        EXPECTED_CLUSTER_COUNT = 2  # first 4 points represent outlier detections
+        generated_detections, _, _ = generate_cluster_data(POINTS_PER_CLUSTER,
+                                                           self.STD_DEV_REGULAR)
+
+        # Run
+        model_ran, detections_in_world = cluster_model.run(generated_detections, False)
+
+        # Test
+        assert model_ran
+        assert (detections_in_world is not None)
+        assert (len(detections_in_world) == EXPECTED_CLUSTER_COUNT)
+
+    def test_detect_skewed_data_five_clusters(self, cluster_model: ClusterEstimation):
+        """
+        data with small distribution but varying number of points per cluster center and
+        random outlier points to simulate false detections
+        """
+        # Setup
+        POINTS_PER_CLUSTER = [1, 1, 1, 1, 20, 100, 100, 100, 100]
+        EXPECTED_CLUSTER_COUNT = 5  # first 4 points represent outlier detections
+        generated_detections, _, _ = generate_cluster_data(POINTS_PER_CLUSTER,
```

Comment:
I am concerned that the library may change its generator in a future version. Having a constant seed only guarantees consistency within a version, not between versions. I think it would be better to have a function that generates a point (or 4 points) given the cluster centres and that guarantees a minimum of distance from them.

Reply:
Good point, function implemented